src/UCommerce.SiteCore.Installer/Steps/EnableSitecoreCompatibilityApp.cs

```diff
@@ -1,27 +1,30 @@
 using System;
 using System.Collections.Specialized;
 using System.IO;
+using System.Threading.Tasks;
 using System.Web.Hosting;
-using Sitecore.Install.Framework;
 using Ucommerce.Installer;
 
 namespace Ucommerce.Sitecore.Installer.Steps
 {
-    public class EnableSitecoreCompatibilityApp : IPostStep
+    public class EnableSitecoreCompatibilityApp : IStep
     {
         private readonly SitecoreVersionChecker _sitecoreVersionChecker;
         private readonly IInstallerLoggingService _sitecoreInstallerLoggingService;
+        private readonly DirectoryInfo _baseDirectory;
 
-        public EnableSitecoreCompatibilityApp(SitecoreVersionChecker sitecoreVersionChecker,
+        public EnableSitecoreCompatibilityApp(SitecoreVersionChecker sitecoreVersionChecker, DirectoryInfo baseDirectory,
             IInstallerLoggingService sitecoreInstallerLoggingService)
         {
             _sitecoreVersionChecker = sitecoreVersionChecker;
             _sitecoreInstallerLoggingService = sitecoreInstallerLoggingService;
+            _baseDirectory = baseDirectory;
         }
 
-        public void Run(ITaskOutput output, NameValueCollection metaData)
+        public async Task Run()
         {
-            string _virtualPathToAppsFolder = "~/sitecore modules/shell/ucommerce/apps";
+            _sitecoreInstallerLoggingService.Information<EnableSitecoreCompatibilityApp>("Enable sitecore compatibility app");
+            string _virtualPathToAppsFolder = $"{_baseDirectory}/sitecore modules/shell/ucommerce/apps";
```

Comment:
We need to get rid of the `HostingEnvironment.MapPath` calls further down.
Since we are not running in a web context we cannot use it.

At the same time instead of doing string handling for the path we should use Path.Combine like in many other steps.